### DISCUSSION TITLE
Voting power is ZERO even though I have tokens at time of snapshot.

### DISCUSSION THREAD

**Discussion by wally000** [snapshot-labs/snapshot#1145] [Q&A] [answered]
Hello,
I am having issued in voting with many snapshot polls. Here for instance I get this one:

<img width="949" alt="image" src="https://user-images.githubusercontent.com/31010273/144028645-d2aff577-6bf4-48ca-9f4e-3225d15f7d46.png">

The snapshot was taken around 30h ago, while my 71 UNI tokens have been in the -430F address for months:

https://etherscan.io/token/0x1f9840a85d5af5bf1d1762f925bdaddc4201f984?a=0xd10a5cb23eef963c7718babf585a97bef28f430f

How come I cannot cast a vote?

As you can see in the screenshot above, my wallet is connected but the zero voting power is shown even before opening the Metamask popup, so it seems an issue with Snapshot reading the tokens amount in my connected wallet.

This is pretty critical and I know friends in my same situation, so please let's work together in solving this.

Thanks.

**Comment by ChaituVR** [comment, marked as answer]
You need to delegate UNI to yourself, undelegated UNI aren't counted for voting power, you can delegate here https://app.uniswap.org/#/vote
